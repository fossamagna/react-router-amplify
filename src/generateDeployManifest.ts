export function generateDeployManifest(reactRouterVersion: string): string {
  const manifest = {
    version: 1,
    framework: {
      name: "react-router",
      version: reactRouterVersion,
    },
    routes: [
      {
        path: "/assets/*",
        target: {
          kind: "Static",
        },
      },
      {
        path: "/*.*",
        target: {
          kind: "Static",
        },
        fallback: {
          kind: "Compute",
          src: "default",
        },
      },
      {
        path: "/*",
        target: {
          kind: "Compute",
          src: "default",
        },
      },
    ],
    computeResources: [
      {
        name: "default",
        runtime: "nodejs20.x",
        entrypoint: "server.mjs",
      },
    ],
  };

  return JSON.stringify(manifest, null, 2);
}
