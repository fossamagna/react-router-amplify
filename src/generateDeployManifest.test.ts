import { describe, expect, test } from "vitest";
import { generateDeployManifest } from "./generateDeployManifest";

describe("generateDeployManifest", () => {
  test("should generate a deploy manifest", () => {
    expect(generateDeployManifest("7.2.0")).toMatchSnapshot();
  });
});
