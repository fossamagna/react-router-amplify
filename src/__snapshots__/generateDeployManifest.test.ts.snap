// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateDeployManifest > should generate a deploy manifest 1`] = `
"{
  "version": 1,
  "framework": {
    "name": "react-router",
    "version": "7.2.0"
  },
  "routes": [
    {
      "path": "/assets/*",
      "target": {
        "kind": "Static"
      }
    },
    {
      "path": "/*.*",
      "target": {
        "kind": "Static"
      },
      "fallback": {
        "kind": "Compute",
        "src": "default"
      }
    },
    {
      "path": "/*",
      "target": {
        "kind": "Compute",
        "src": "default"
      }
    }
  ],
  "computeResources": [
    {
      "name": "default",
      "runtime": "nodejs20.x",
      "entrypoint": "server.mjs"
    }
  ]
}"
`;
